android-components/components/feature/search/src/main/java/mozilla/components/feature/search/telemetry/SearchProviderModel.kt

```diff
@@ -4,48 +4,74 @@
 
 package mozilla.components.feature.search.telemetry
 
+import kotlinx.serialization.SerialName
+import kotlinx.serialization.Serializable
+
 /**
  * All data needed to identify ads of a particular provider.
  *
- * @property name provider name.
- * @property regexp regular expression used to match the provider.
- * @property queryParam name of the query parameter for the user's search string.
- * @property codeParam name of the query parameter for the partner code.
- * @property codePrefixes array of prefixes (or complete values) to match against
+ * @property taggedCodes array of prefixes (or complete values) to match against
  * the partner code parameters in the url.
- * @property followOnParams array of query parameter names that are used when a follow-on search occurs.
+ * @property telemetryId provider name e.g. "google", "duckduckgo".
+ * @property organicCodes array of partner codes to match against the parameters in the url.
+ * Matching these codes will report the SERP as organic:<partner code>, which means the search
+ * was performed organically rather than through a SAP.
+ * @property searchPageRegexp regular expression used to match the provider.
+ * @property queryParamNames list of names of the query parameters for the user's search string.
+ * @property codeParamName name of the query parameter for the partner code.
+ *  @property followOnParamNames array of query parameter names that are used when a follow-on search occurs.
  * @property extraAdServersRegexps array of regular expressions that match URLs of potential ad servers.
  * @property followOnCookies array of cookie details that are used to identify follow-on searches.
+ * @property expectedOrganicCodes array of partner codes to match against the parameters in the url.
+ * Matching these codes will report the SERP as organic:none which means the user has done a search
+ * through the search engine's website rather than through SAP.
+ * @property adServerAttributes an array of strings that potentially match data-attribute keys of anchors.
  */
-internal data class SearchProviderModel(
-    val name: String,
-    val regexp: Regex,
-    val queryParam: String,
-    val codeParam: String,
-    val codePrefixes: List<String>,
-    val followOnParams: List<String>,
+@Serializable
+data class SearchProviderModel(
+    val schema: Long,
+    val taggedCodes: List<String>,
+    @SerialName("telemetryId")
+    val telemetryId: String,
+    val organicCodes: List<String>?,
+    val searchPageRegexp: Regex,
+    val queryParamNames: List<String>?,
+    @SerialName("codeParamName")
+    val codeParamName: String,
+    @SerialName("followOnParamNames")
+    val followOnParamNames: List<String>?,
```

Comment:
Are all of these properties nullable now because we might fail parsing the JSON, or because they are actually optional?